hatchet_sdk/clients/dispatcher/action_listener.py

```diff
@@ -55,8 +56,9 @@ def __post_init__(self):
                 self.labels[key] = WorkerLabels(strValue=str(value))
 
 
-@dataclass
-class Action:
+class Action(BaseModel):
+    model_config = ConfigDict(arbitrary_types_allowed=True)
```

Comment:
drive by - I think making this a pydantic model makes sense b/c it allows for what I view as a little bit nicer validation of the metadata coming in, + there's validation of the other types (side note: I think mypy would help a lot here too)